tools/karl.cpp

```diff
@@ -1038,6 +1068,114 @@ void handle_arguments (int argc, char ** argv)
   }
 }
 
+
+// recursively loads a config file(s) into a pased in string vector.
+// current file recurse limit is adjustable.
+bool load_config_file(std::vector<char *>& argv_config_files, 
+                      std::string full_path, int file_cnt, int flag_cnt)
+{
+  std::string flag;
+  std::string param;
+  std::string flag_param;
+  std::string config_file;  
+ 
+  // load the a karl config file
+  std::ifstream file(full_path.c_str()); 
+  if (!file) 
+  {
+      madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                            "Unable to open karl config file: %s\n", 
+                            full_path); 
+        
+      return false;      
+  }
+  else
+  {
+    if (flag_cnt == 0)
+    {
+      // load progam path for first arg
+      std::string obPath("karlconfig");
+      argv_config_files.push_back(nullptr);
+      argv_config_files[flag_cnt] = new char[obPath.size() + 1];
+      obPath.copy(argv_config_files[flag_cnt], obPath.size(), 0);
+      argv_config_files[flag_cnt][obPath.size()] = 0; // null string terminator 
+      flag_cnt++; 
+    } 
+
+    madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                          "Found karl config file: %s\n", 
+                          full_path);                   
+       
+    // read each line of the text formatted file in, each line contains 
+    // one flag followed by a space then the param if there is a parameter
+    while (std::getline(file, flag_param))
+    {
+      flag = flag_param.substr(0, flag_param.find_first_of(" "));
+      if (flag_param.find_first_of(" ") == std::string::npos)
+      {
+        param = "";
+      }
+      else
+      {
+        param = flag_param.substr(flag_param.find_first_of(" ") + 1,
+        flag_param.length() - (flag_param.find_first_of(" ") + 1)); 
+      }      
+      madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                            "Flag: %s, Param: %s\n", 
+                            flag, param);         
+      
+      if ((flag == "-cf" || flag == "--config-file") && file_cnt >= 0)
```

Comment:
A roger on that.

Reply:
Just a note making this change eliminates the need for load_config_file() to be recursive.
true?
Cause then it just processes that one file but in a sense handle_arguments() is made recursive since it is called from load_config_file(). Also would fix above issue ttoo.
Before I proceed I want to make sure you agree